### DISCUSSION TITLE
Overriding font settings for specific apps?

### DISCUSSION THREAD

**Discussion by Vandino86** [lxqt/lxqt#2654] [Q&A]
I was wondering if there was a way to use a font or font size that's different from what's currently set in LXQt Appearance Configuration --> Font for a specific application. For example, I'm using DejaVu Sans size 11 (the defaults) for LXQt, but I want to use a smaller and/or different font for Firefox. However, I want to keep using DejaVu Sans size 11 for all my other apps. Is there a way I can change the font just for Firefox without changing it for all of LXQt? The font settings within Firefox (about:preferences --> Fonts) change the fonts for webpages rather than the fonts for the UI (tabs, bookmarks toolbar, hamburger menu, etc.), which is what I want.

**Comment by Bluey26** [comment]
Hi, it seems to be possible to override the font choice for certain applications. For some reason, it does not seem to work in Qt apps, but it worked for GTK apps (Firefox is a GTK app).

I found the information in reddit (https://reddit.com/r/archlinux/comments/dim7dt/fontconfig_can_i_disable_a_specific_font_for_an/) and adapted it somehow to match your needs.

So far, this override all the _fonts families_ for the specified program. I tried it with several programs and i was able to change the font of the GTK programs i tried.

You need to create the `~/.config/fontconfig/fonts.conf` file, and add the following inside` <fontconfig> </fontconfig>` :

```
<match>
  <test name="prgname"><string>firefox</string></test>
  <edit mode="assign" name="family" binding="same">
   <string>Monospace</string>
  </edit>
</match>
```

This will override the font for firefox, setting the font to be Monospace.

If you want a better result, you could modify several font families, making the text to being displayed better in firefox (if you see the Firefox options, there are several fonts, not just one). For this, you would need to something like this:

```
 <match target="pattern">
  <test name="family" qual="any">
   <string>monospace</string>
  </test>
  <edit mode="assign" name="family" binding="same">
   <string>Liberation Mono</string>
  </edit>
 </match>
```
 
Unfortunately i was not able to make it work.

**Reply by tsujan** [reply]
This is not an answer to the question.

Editing `~/.config/fontconfig/fonts.conf`  isn't recommended. You could make a conf file like `~/.config/fontconfig/conf.d/0-custom.conf` instead.

**Reply by Vandino86** [reply]
If it helps, I'm specifically trying to decrease the font size from 11 to 10 or 9. I made a file `~/.config/fontconfig/conf.d/0-custom.conf` and put the following in it. Unfortunately, it's not working for me, even without `binding="same"`.

```
<?xml version='1.0'?>
<!DOCTYPE fontconfig SYSTEM 'fonts.dtd'>
<fontconfig>
 <match>
  <test name="prgname"><string>firefox</string></test>
  <edit mode="assign" name="size" binding="same">
   <double>10</double>
  </edit>
 </match>
</fontconfig>
```
